### DISCUSSION TITLE
Dynamic update of README.md?

### DISCUSSION THREAD

**Discussion by hakuseki** [quarkusio/quarkus#31468] [Q&A]
Hi

Are there anything that can update the README.md file **after** changes made to the project?

Since the README-file contains information and resources for any developer to dive in to, but it is generated upon creation of project, it would be nice to be able to update it base don pom changes, dependencies and new/removed extensions.

Thoughts, suggestions, tips?

Thx

M

**Comment by maxandersen** [comment]
Not following what you are expecting to be available ?

Are you asking if when quarkus extension add is used that there is something done to the readme file?

**Reply by hakuseki** [reply]
Well the file itself contains a lot of information about the project, such as extensions and stuff. 

It is generated automatically when using the Quarkus project wizard but is never then changed when you add or remove extensions or change other values in your project. 

After a while the file is filled with useless information or wrong information. 

I'm just asking if there is a way to "regenerate" the file so it reflects the project more accurately. 
It will also remove the need for manually updating the file. 

Basically I'm just lazy… 🤓

**Reply by maxandersen** [reply]
Ok - so no. We can't assume the file is untouched so we can't safely edit it. 

In theory we could somewhat regenerate it but I would say you are better of making your own readme file. 

The readme file is for your project. 

having just info about the subparts used in your project doesn't help that much does it?

**Reply by hakuseki** [reply]
Of course it would be overwritten, that is a prerequisite, otherwise would be more or less impossible. 

I personally find this file useful especially if you are to handover the project to someone else later. 

The readme I create holds all info about the functionality and everything and it is created and maintained under src as Asciidoc. 

Just a thought…

Thanks 🙏
